include/flatbuffers/flatbuffers.h

```diff
@@ -960,9 +960,11 @@ FLATBUFFERS_FINAL_CLASS
   }
 
   /// @brief Store a string in the buffer, which can contain any binary data.
-  /// @param[in] str A const reference to a std::string to store in the buffer.
+  /// @param[in] str A const reference to a std::string like type with support
+  /// of T::c_str and T::length() to store in the buffer.
   /// @return Returns the offset in the buffer where the string starts.
-  Offset<String> CreateString(const std::string &str) {
+  template<typename T>
+  Offset<String> CreateString(const T &str) {
```

Comment:
Thanks for checking! For now, please keep the "dead code" std::string version in there, to be sure. Also please move the template to after all the non-templated versions, just in case that matters.